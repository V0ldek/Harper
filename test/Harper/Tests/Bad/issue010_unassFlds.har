value List a = {
  variant Empty = { };
  
  variant NonEmpty = {
    data = {
      head :: a;
      tail :: List a;
    }
  };
};

cons :: a -> List a -> List a;
cons x xs = val NonEmpty { head = x, tail = xs };

empty :: List a;
empty = val Empty {};

main :: List Integer;
main = {
  (list :: List Integer) = val NonEmpty { head = 5 };
  return list;
};