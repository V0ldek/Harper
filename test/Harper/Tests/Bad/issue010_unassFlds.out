Error: all fields must be assigned during value construction. Unassigned fields for 'List.NonEmpty' are: `tail`.
During evaluation of:
 val NonEmpty { head = 5 }

Located at line 20 column 28

Execution terminated with an error: type error.